src/TimingData.cpp

```diff
@@ -15,6 +15,17 @@ static const int INVALID_INDEX = -1;
 
 TimingSegment* GetSegmentAtRow( int iNoteRow, TimingSegmentType tst );
 
+// According to The C++ Programming Language 3rd Ed., decreasing the size
```

Comment:
Okay, just confirmed with a C++ channel. Pre-c++11, this trick is needed. Now, there is a member function that can handle it without creating a new instance. Look into [shrink_to_fit](http://en.cppreference.com/w/cpp/container/vector/shrink_to_fit).